### PR TITLE
Update ecdsa.h

### PR DESCRIPTION
**Pull request by RosZad**
Minor documentation improvements:
*Standardized file brief description.
*Separated return statements.
*Reordered tags within documentation blocks so that params and returns are last in block.
*p_rng descriptions changed from "parameter" to "context".

### CODE REVIEW

File: include/mbedtls/ecdsa.h [hunk 1]
```diff
@@ -109,17 +120,9 @@ int mbedtls_ecdsa_sign( mbedtls_ecp_group *grp, mbedtls_mpi *r, mbedtls_mpi *s,
...
- * \note            If the bitlength of the message hash is larger than the
- *                  bitlength of the group order, then the hash is truncated as
- *                  defined in <em>Standards for Efficient Cryptography Group
- *                  (SECG): SEC1 Elliptic Curve Cryptography</em>, section
- *                  4.1.3, step 5.
- *
- * \return          \c 0 on success,
- *                  or an \c MBEDTLS_ERR_ECP_XXX or \c MBEDTLS_MPI_XXX
+ * \return          \c 0 on success.
+ * \return          or an \c MBEDTLS_ERR_ECP_XXX or \c MBEDTLS_MPI_XXX
```
Comment (hanno-arm) [inline include/mbedtls/ecdsa.h]: Blocker: Replace `or an` by `An` as above.

File: include/mbedtls/ecdsa.h [hunk 2]
```diff
@@ -130,25 +133,25 @@ int mbedtls_ecdsa_sign_det( mbedtls_ecp_group *grp, mbedtls_mpi *r, mbedtls_mpi
...
- *                  bitlength of the group order, then the hash is truncated as
- *                  defined in <em>Standards for Efficient Cryptography Group
- *                  (SECG): SEC1 Elliptic Curve Cryptography</em>, section
- *                  4.1.4, step 3.
- *
- * \return          \c 0 on success,
- *                  #MBEDTLS_ERR_ECP_BAD_INPUT_DATA if signature is invalid,
- *                  or an \c MBEDTLS_ERR_ECP_XXX or \c MBEDTLS_MPI_XXX
+ * \return          \c 0 on success.
+ * \return          #MBEDTLS_ERR_ECP_BAD_INPUT_DATA if signature is invalid.
```
Comment (hanno-arm) [inline include/mbedtls/ecdsa.h]: Suggestion: `if the signature` instead of `if signature`.

File: include/mbedtls/ecdsa.h [hunk 3]
```diff
@@ -241,11 +235,20 @@ int mbedtls_ecdsa_write_signature( mbedtls_ecdsa_context *ctx, mbedtls_md_type_t
...
- * \return          \c 0 on success,
- *                  or an \c MBEDTLS_ERR_ECP_XXX, \c MBEDTLS_ERR_MPI_XXX or
- *                  \c MBEDTLS_ERR_ASN1_XXX error code on failure.
- *
  * \see             ecp.h
+ *
+ * \deprecated      Superseded by mbedtls_ecdsa_write_signature() in 2.0.0
+ *
+ * \param ctx       The ECDSA context.
+ * \param hash      The Message hash.
```
Comment (hanno-arm) [inline include/mbedtls/ecdsa.h]: Please change `Message` to `message` (pre-existing).

File: include/mbedtls/ecdsa.h [hunk 4]
```diff
@@ -1,9 +1,10 @@
 /**
  * \file ecdsa.h
  *
- * \brief The Elliptic Curve Digital Signature Algorithm (ECDSA).
+ * \brief This file contains ECDSA definitions and functions.
```
Comment (AndrzejKurek) [inline include/mbedtls/ecdsa.h]: I feel like this is a good place to have the full name with an abbreviation in the parentheses.

File: include/mbedtls/ecdsa.h (hunk 1 above)
Comment (RosZad) [reply include/mbedtls/ecdsa.h]: done

File: include/mbedtls/ecdsa.h (hunk 3 above)
Comment (RosZad) [reply include/mbedtls/ecdsa.h]: fixed

File: include/mbedtls/ecdsa.h (hunk 2 above)
Comment (RosZad) [reply include/mbedtls/ecdsa.h]: changed

File: include/mbedtls/ecdsa.h (hunk 4 above)
Comment (RosZad) [reply include/mbedtls/ecdsa.h]: Disagree. The brief description refers to the content of the file. The detailed description contains all other information, including any standard description or reference. Having the full name in the brief makes it harder to read with comprehension.

File: include/mbedtls/ecdsa.h [hunk 5]
```diff
@@ -241,11 +236,20 @@ int mbedtls_ecdsa_write_signature( mbedtls_ecdsa_context *ctx, mbedtls_md_type_t
  *                  (SECG): SEC1 Elliptic Curve Cryptography</em>, section
  *                  4.1.3, step 5.
  *
- * \return          \c 0 on success,
- *                  or an \c MBEDTLS_ERR_ECP_XXX, \c MBEDTLS_ERR_MPI_XXX or
- *                  \c MBEDTLS_ERR_ASN1_XXX error code on failure.
- *
  * \see             ecp.h
+ *
+ * \deprecated      Superseded by mbedtls_ecdsa_write_signature() in 2.0.0
```
Comment (sbutcher-arm) [inline include/mbedtls/ecdsa.h]: Pre-existing, but I don't think 'in 2.0.0' obviously means to the reader, Mbed TLS version 2.0.0.

I suggest we change the text to 'Superseded by mbedtls_ecdsa_write_signature() in Mbed TLS version 2.0 and later'.

Comment (RosZad) [reply include/mbedtls/ecdsa.h]: Agreed and modified

**Comment by RosZad** [comment]
@gilles-peskine-arm - can you please advise what the conflicting PR number is?

**Comment by gilles-peskine-arm** [comment]
The conflict comes from #1543

**Pull request merged**